projects/sandbox/datagen/datagen/utils/injection.py

```diff
@@ -91,6 +95,67 @@ def generate_gw(
     return signals
 
 
+def generate_gw_bns(
+    sample_params: Dict[List, str],
+    minimum_frequency: float,
+    reference_frequency: float,
+    sample_rate: float,
+    waveform_duration: float,
+    waveform_approximant: str,
+    detector_frame_prior: bool = False,
+):
+    padding = 1
+    if not detector_frame_prior:
+        sample_params = convert_to_detector_frame(sample_params)
+
+    sample_params = [
+        dict(zip(sample_params, col)) for col in zip(*sample_params.values())
+    ]
+
+    n_samples = len(sample_params)
+
+    waveform_generator = WaveformGenerator(
+        duration=waveform_duration + padding,
+        sampling_frequency=sample_rate,
+        frequency_domain_source_model=lal_binary_neutron_star,
+        parameter_conversion=convert_to_lal_binary_neutron_star_parameters,
+        waveform_arguments={
+            "waveform_approximant": waveform_approximant,
+            "reference_frequency": reference_frequency,
+            "minimum_frequency": minimum_frequency,
+        },
+    )
+
+    logging.info("Generating BNS waveforms     : {}".format(n_samples))
+
+    waveform_size = int(sample_rate * waveform_duration)
+    num_pols = 2
+    signals = np.zeros((n_samples, num_pols, waveform_size))
+
+    for i, p in enumerate(sample_params):
+        polarizations = waveform_generator.time_domain_strain(p)
+        polarization_names = sorted(polarizations.keys())
+        polarizations = np.stack(
+            [polarizations[p] for p in polarization_names]
+        )
+
+        # just shift the coalescence to the left by 200 datapoints
+        # to cancel wraparound in the beginning
+        dt = -200
```

Comment:
Again, another magic number: why is this 200? Is there a first principles motivation for this?

Reply:
Please see above